be/src/exec/base_scanner.cpp

```diff
@@ -0,0 +1,163 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+
+#include "base_scanner.h"
+
+#include "runtime/descriptors.h"
+#include "runtime/mem_tracker.h"
+#include "runtime/raw_value.h"
+#include "runtime/tuple.h"
+#include "runtime/runtime_state.h"
+
+namespace doris {
+
+BaseScanner::BaseScanner(RuntimeState* state, const TBrokerScanRangeParams& params, ScannerCounter* counter) :
+        _state(state), _params(params), _counter(counter),
+        _src_tuple(nullptr),
+        _src_tuple_row(nullptr),
+#if BE_TEST
+        _mem_tracker(new MemTracker()),
+        _mem_pool(_mem_tracker.get()),
+#else
+        _mem_tracker(new MemTracker(-1, "Broker Scanner", state->instance_mem_tracker())),
+        _mem_pool(_state->instance_mem_tracker()),
+#endif
+        _dest_tuple_desc(nullptr),
+        _strict_mode(false) {
+}
+
+Status BaseScanner::init_expr_ctxes() {
```

Comment:
This is base class's open, and in child class call this BaseScanner::open()
```suggestion
Status BaseScanner::open() {
```